ui/app/helpers/utils/i18n-helper.js

```diff
@@ -39,13 +40,25 @@ export const getMessage = (localeCode, localeMessages, key, substitutions) => {
   }
   const entry = localeMessages[key]
   let phrase = entry.message
+
+  const hasSubstitutions = Boolean(substitutions && substitutions.length)
+  const hasReactSubstitutions = hasSubstitutions &&
+    substitutions.some((element) => typeof element === 'function' || typeof element === 'object')
+
   // perform substitutions
-  if (substitutions && substitutions.length) {
-    substitutions.forEach((substitution, index) => {
-      const regex = new RegExp(`\\$${index + 1}`, 'g')
-      phrase = phrase.replace(regex, substitution)
+  if (hasSubstitutions) {
+    const parts = phrase.split(/(\$\d)/g)
+
+    const substitutedParts = parts.map((part) => {
+      const subMatch = part.match(/\$(\d)/)
+      return subMatch ? substitutions[Number(subMatch[1]) - 1] : part
```

Comment:
Ideally we'd have some error handling here if `substitutions[index]` was undefined. Better to throw an error than show "undefined" to the user.